### ISSUE TITLE
Migration error: sproutreports.users

### ISSUE THREAD

**Issue by darylknight** [barrelstrength/craft-sprout-forms#522]
### Description (screenshots, logs, ...)
Just tried updating Sprout Forms from 3.11.7 to latest (Craft 3.4.30) and got this migration error.

```
Database Exception: SQLSTATE[22007]: Invalid datetime format: 1292 Truncated incorrect INTEGER value: 'sproutreports.users'
The SQL being executed was: ALTER TABLE craft_sproutreports_reports CHANGE dataSourceId dataSourceId int(11)

Migration: barrelstrength\sproutforms\migrations\m200520_000002_update_datasourceId_column_type_sproutforms

Output:

> alter column dataSourceId in table {{%sproutreports_reports}} to integer ...Exception: SQLSTATE[22007]: Invalid datetime format: 1292 Truncated incorrect INTEGER value: 'sproutreports.users'
The SQL being executed was: ALTER TABLE craft_sproutreports_reports CHANGE dataSourceId dataSourceId int(11) (/Users/dknight/Sites/gdb/vendor/yiisoft/yii2/db/Schema.php:674)
#0 /Users/dknight/Sites/gdb/vendor/yiisoft/yii2/db/Command.php(1298): yii\db\Schema->convertException(Object(PDOException), 'ALTER TABLE cr...')
#1 /Users/dknight/Sites/gdb/vendor/yiisoft/yii2/db/Command.php(1093): yii\db\Command->internalExecute('ALTER TABLE cr...')
#2 /Users/dknight/Sites/gdb/vendor/yiisoft/yii2/db/Migration.php(420): yii\db\Command->execute()
#3 /Users/dknight/Sites/gdb/vendor/barrelstrength/sprout-base-reports/src/migrations/m200520_000002_update_datasourceId_column_type.php(25): yii\db\Migration->alterColumn('{{%sproutreport...', 'dataSourceId', Object(craft\db\mysql\ColumnSchemaBuilder))
#4 /Users/dknight/Sites/gdb/vendor/barrelstrength/sprout-forms/src/migrations/m200520_000002_update_datasourceId_column_type_sproutforms.php(18): barrelstrength\sproutbasereports\migrations\m200520_000002_update_datasourceId_column_type->safeUp()
#5 /Users/dknight/Sites/gdb/vendor/craftcms/cms/src/db/Migration.php(52): barrelstrength\sproutforms\migrations\m200520_000002_update_datasourceId_column_type_sproutforms->safeUp()
#6 /Users/dknight/Sites/gdb/vendor/craftcms/cms/src/db/MigrationManager.php(233): craft\db\Migration->up(true)
#7 /Users/dknight/Sites/gdb/vendor/craftcms/cms/src/db/MigrationManager.php(153): craft\db\MigrationManager->migrateUp(Object(barrelstrength\sproutforms\migrations\m200520_000002_update_datasourceId_column_type_sproutforms))
#8 /Users/dknight/Sites/gdb/vendor/craftcms/cms/src/services/Updates.php(233): craft\db\MigrationManager->up()
#9 /Users/dknight/Sites/gdb/vendor/craftcms/cms/src/controllers/BaseUpdaterController.php(510): craft\services\Updates->runMigrations(Array)
#10 /Users/dknight/Sites/gdb/vendor/craftcms/cms/src/controllers/UpdaterController.php(203): craft\controllers\BaseUpdaterController->runMigrations(Array, 'restore-db')
#11 [internal function]: craft\controllers\UpdaterController->actionMigrate()
#12 /Users/dknight/Sites/gdb/vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array(Array, Array)
#13 /Users/dknight/Sites/gdb/vendor/yiisoft/yii2/base/Controller.php(157): yii\base\InlineAction->runWithParams(Array)
#14 /Users/dknight/Sites/gdb/vendor/craftcms/cms/src/web/Controller.php(178): yii\base\Controller->runAction('migrate', Array)
#15 /Users/dknight/Sites/gdb/vendor/yiisoft/yii2/base/Module.php(528): craft\web\Controller->runAction('migrate', Array)
#16 /Users/dknight/Sites/gdb/vendor/craftcms/cms/src/web/Application.php(291): yii\base\Module->runAction('updater/migrate', Array)
#17 /Users/dknight/Sites/gdb/vendor/craftcms/cms/src/web/Application.php(646): craft\web\Application->runAction('updater/migrate')
#18 /Users/dknight/Sites/gdb/vendor/craftcms/cms/src/web/Application.php(243): craft\web\Application->_processUpdateLogic(Object(craft\web\Request))
#19 /Users/dknight/Sites/gdb/vendor/yiisoft/yii2/base/Application.php(386): craft\web\Application->handleRequest(Object(craft\web\Request))
#20 /Users/dknight/Sites/gdb/public/index.php(21): yii\base\Application->run()
#21 /Users/dknight/.composer/vendor/laravel/valet/server.php(158): require('/Users/darylkni...')
#22 {main}
```

### Additional info

- Craft version: 3.4.30
- PHP version: 7.4.8
- Database driver & version: MySQL 5.5.5

<img src="https://frontapp.com/assets/img/favicons/favicon-32x32.png" height="16" width="16" alt="Front logo" /> [Front conversations](https://app.frontapp.com/open/top_3emkt)

**Comment by darylknight** [comment]
Additionally, trying to update from 3.11.7 to 3.11.8 gives this error:

```
Error: Class 'barrelstrength\sproutbasereports\migrations\m200520_000001_update_user_datasourceid' not found

Migration: barrelstrength\sproutforms\migrations\m200520_000001_update_user_datasourceid_sproutforms

Output:

Exception: Class 'barrelstrength\sproutbasereports\migrations\m200520_000001_update_user_datasourceid' not found (/Users/dknight/Sites/gdb/vendor/barrelstrength/sprout-forms/src/migrations/m200520_000001_update_user_datasourceid_sproutforms.php:15)
#0 /Users/dknight/Sites/gdb/vendor/craftcms/cms/src/db/Migration.php(52): barrelstrength\sproutforms\migrations\m200520_000001_update_user_datasourceid_sproutforms->safeUp()
#1 /Users/dknight/Sites/gdb/vendor/craftcms/cms/src/db/MigrationManager.php(233): craft\db\Migration->up(true)
#2 /Users/dknight/Sites/gdb/vendor/craftcms/cms/src/db/MigrationManager.php(153): craft\db\MigrationManager->migrateUp(Object(barrelstrength\sproutforms\migrations\m200520_000001_update_user_datasourceid_sproutforms))
#3 /Users/dknight/Sites/gdb/vendor/craftcms/cms/src/services/Updates.php(233): craft\db\MigrationManager->up()
#4 /Users/dknight/Sites/gdb/vendor/craftcms/cms/src/controllers/BaseUpdaterController.php(510): craft\services\Updates->runMigrations(Array)
#5 /Users/dknight/Sites/gdb/vendor/craftcms/cms/src/controllers/UpdaterController.php(203): craft\controllers\BaseUpdaterController->runMigrations(Array, 'restore-db')
#6 [internal function]: craft\controllers\UpdaterController->actionMigrate()
#7 /Users/dknight/Sites/gdb/vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array(Array, Array)
#8 /Users/dknight/Sites/gdb/vendor/yiisoft/yii2/base/Controller.php(157): yii\base\InlineAction->runWithParams(Array)
#9 /Users/dknight/Sites/gdb/vendor/craftcms/cms/src/web/Controller.php(178): yii\base\Controller->runAction('migrate', Array)
#10 /Users/dknight/Sites/gdb/vendor/yiisoft/yii2/base/Module.php(528): craft\web\Controller->runAction('migrate', Array)
#11 /Users/dknight/Sites/gdb/vendor/craftcms/cms/src/web/Application.php(291): yii\base\Module->runAction('updater/migrate', Array)
#12 /Users/dknight/Sites/gdb/vendor/craftcms/cms/src/web/Application.php(646): craft\web\Application->runAction('updater/migrate')
#13 /Users/dknight/Sites/gdb/vendor/craftcms/cms/src/web/Application.php(223): craft\web\Application->_processUpdateLogic(Object(craft\web\Request))
#14 /Users/dknight/Sites/gdb/vendor/yiisoft/yii2/base/Application.php(386): craft\web\Application->handleRequest(Object(craft\web\Request))
#15 /Users/dknight/Sites/gdb/public/index.php(21): yii\base\Application->run()
#16 /Users/dknight/.composer/vendor/laravel/valet/server.php(158): require('/Users/darylkni...')
#17 {main}
```

**Comment by krizzzly** [comment]
this is related to this one: https://github.com/barrelstrength/craft-sprout-reports/issues/95#issuecomment-667912410

**Comment by BenParizek** [comment]
@darylknight @krizzzly Looking into this. 

We're currently focused on polishing up a major release that adds Craft 3.5 support and addresses several migration scenarios that have popped up over the last months. Due to an architectural decision I made a while back, some of our migration scenarios have become more tedious than they need to be and we're working on getting that resolved. I'll update this thread when I have more.

**Comment by OscarBarrett** [comment]
Manually skipping the migration appears to work. (i.e. add `m200520_000002_update_datasourceId_column_type_sproutforms` to the migrations table manually)

**Comment by BenParizek** [comment]
This should now be fixed if you upgrade to the latest (v3.13.4).
